pkg/distributor/push.go

```diff
@@ -198,19 +290,19 @@ func calculateRetryAfter(retryAttemptHeader string, baseSeconds int, maxBackoffE
 	return strconv.FormatInt(delaySeconds, 10)
 }
 
-// toHTTPStatus converts the given error into an appropriate HTTP status corresponding
-// to that error, if the error is one of the errors from this package. Otherwise, an
+// toGRPCHTTPStatus converts the given error into an appropriate GRPC and HTTP status corresponding
+// to that error, if the error is one of the errors from this package. Otherwise, codes.Internal and
 // http.StatusInternalServerError is returned.
-func toHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrides) int {
+func toGRPCHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrides) (codes.Code, int) {
 	if errors.Is(pushErr, context.DeadlineExceeded) {
```

Comment:
nit: this conditionally isn't really needed since this is the default return at the bottom of the function.

Reply:
let me do it in a separate PR, since this is not related to the otlp change, don't want to break the code in push.